pkg/services/sqlstore/alert.go

```diff
@@ -339,3 +342,47 @@ func GetAlertStatesForDashboard(query *m.GetAlertStatesForDashboardQuery) error
 
 	return err
 }
+
+func GetMissingAlerts(query *m.GetMissingAlertsQuery) error {
+
+	//Get current timestamp
+	var ts int64 = -1
+	err := inTransaction(func(sess *xorm.Session) error {
+		results, err := sess.Query("select " + dialect.CurrentTimeToRoundMinSql() + " as ts ")
+		if err != nil {
+			sqlog.Error("Failed to get timestamp", "error", err)
+			return err
+		}
+		ts, err = strconv.ParseInt(string(results[0]["ts"]), 10, 64)
+		if err != nil {
+			sqlog.Error("Failed to get timestamp", "error", err)
+			return err
+		}
+		return nil
+	})
+	if err != nil {
+		sqlog.Error("Transaction failed", "error", err)
+		return err
+	}
+
+	//Get all alerts
+	cmd := &m.GetAllAlertsQuery{}
+	if err := bus.Dispatch(cmd); err != nil {
+		sqlog.Error("Could not load alerts", "error", err)
+		return err
+	}
+
+	//Find Missing alerts
+	var evalTime int64
+	var frequency int64
+	missedAlerts := make([]*m.Alert, 0)
```

Comment:
There are two competing goals here
1. Allow the normal processing in each node to evaluate all alerts within 1 min interval (and thus not requiring missing alert runs)
2. Detect and evaluate any missing alert but not delay "next" normal processing on the node chosen for evaluating missing alert - It might so happen that the node chosen for evaluating missing alert wouldn't be able to participate in next normal run and thus, create more missing alerts.

Operationally speaking, normal processing might have higher significance (since it will be using latest data points), hence it might make sense to set a configurable upper bound on the length of the array for missing alerts.